airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -114,59 +140,46 @@ export const ReplicationView: React.FC<ReplicationViewProps> = ({ onAfterSaveSch
     if (activeUpdatingSchemaMode) {
       setActiveUpdatingSchemaMode(false);
     }
-
-    formikHelpers?.resetForm({ values });
   };
 
-  const openResetDataModal = (values: ValuesProps) => {
-    openConfirmationModal({
-      title: "connection.updateSchema",
-      text: "connection.updateSchemaText",
-      submitButtonText: "connection.updateSchema",
-      submitButtonDataId: "refresh",
-      onSubmit: async () => {
-        await onSubmit(values);
-        closeConfirmationModal();
-      },
-    });
-  };
-
-  const onSubmitForm = async (values: ValuesProps) => {
-    if (activeUpdatingSchemaMode) {
-      openResetDataModal(values);
+  const onSubmitForm = async (values: ValuesProps): Promise<void | ConnectionFormSubmitResult> => {
+    // Detect whether the catalog has any differences compared to the original one.
+    // This could be due to user changes (e.g. in the sync mode) or due to new/removed
+    // streams due to a "refreshed source schema".
+    const hasCatalogChanged = !equal(values.syncCatalog, initialConnection.syncCatalog);
+    // Whenever the catalog changed show a warning to the user, that we're about to reset their data.
+    // Given them a choice to opt-out in which case we'll be sending skipRefresh: true to the update
+    // endpoint.
+    if (hasCatalogChanged) {
+      const stateType = await connectionService.getStateType(connectionId);
+      const result = await openModal<boolean>({
+        title: formatMessage({ id: "connection.resetModalTitle" }),
+        content: (props) => <ResetWarningModal {...props} stateType={stateType} />,
+      });
+      if (result.type === "canceled") {
+        return {
+          submitCancelled: true,
+        };
+      }
+      // Save the connection taking into account the correct skipRefresh value from the dialog choice.
+      await saveConnection(values, !result.reason);
     } else {
-      await onSubmit(values);
+      // The catalog hasn't changed. We don't need to ask for any confirmation and can simply save.
+      await saveConnection(values, true);
     }
   };
 
-  const onEnterRefreshCatalogMode = async () => {
+  const onRefreshSourceSchema = async () => {
+    setSaved(false);
```

Comment:
:information_source: This solves some issue that the "This connection was saved" reappears again after an error vanished, despite no new save has happened.